SCClassLibrary/Common/Collections/Dictionary.sc

```diff
@@ -560,12 +560,39 @@ IdentityDictionary : Dictionary {
 		^this.superPerformList(\doesNotUnderstand, selector, args);
 	}
 
-		// Quant support.
-		// The Quant class assumes the quant/phase/offset scheduling model.
-		// If you want a different model, you can write a dictionary like so:
-		// (nextTimeOnGrid: { |self, clock| ... calculate absolute beat number here ... },
-		//	parameter: value, parameter: value, etc.)
-		// If you leave out the nextTimeOnGrid function, fallback to quant/phase/offset.
+	doesNotUnderstandWithKeys {|selector, argsArray, keywordArgsAsPairs|
+		if(know.not){
+			^this.superPerformList(\doesNotUnderstandWithKeys, selector, argsArray, keywordArgsAsPairs )
+		};
+
+		this[selector] !? {|f|
+			^f.functionPerformList(
+                \value,
+                 f.def.makePerformableArray([this] ++ argsArray, keywordArgsAsPairs.asEvent)
+             )
+		};
+
+		// Unlike in doesNotUnderstand, we don't convert to a setter as this isn't
+		// possible with keyword args.
+
+		// If the keyword arg names don't match this will fail.
+		// Note how the selector is passed in here.
+		this[\forward] !? {|f|
+			^f.functionPerformList(
```

Comment:
No, that would change how object prototyping calls methods and potentially break stuff.

Reply:
ah sorry, sure. I meant `^func.performWith(\value, [this] ++ argsArray, keywordArgsAsPairs.asEvent)`

Reply:
No that still won't work because it needs to go through `functionPerformList` which is how classes currently opt into object prototyping.

Reply:
I thought that it may be inefficient to calculate the array when none is needed. So a method like `functionPerformWith` would be logical, which would simply return `this` for objects and call `valueWith` for functions.

But, I suppose it is a rare case anyhow that someone passes arguments when simply accessing an object in a prototype.

Like: 
```
p = (finch: [1, 2, 3]);
p.finch; // yes
p.finch(freq: 900); // rather not
```

Reply:
I think there's a nicely idiomatic way to handle this. It's not unlikely someone wants to `p.finch` to sometjing else, or?
```
p.finch_((freq: 900));
```
equivant to
```
p.finch = (freq: 900);
```
But that syntax doesn't cause any conflict, does it?

Reply:
That's not a keyword argument, and should already work? Away from computer, but I'll test later.

Reply:
> p.finch(freq: 900); 

Is this to demonstrate a case where the user just wants to access the value?

What you've written throws an error because the keyword does not exist as a selector on Array (I think).

I don't think it ever makes sense to use a keyword when you want to access something.

Reply:
I imagine the case where the message `finch` would be sent to different prototype objects (dictionaries), and some of them have a function in finch to generate a new object, taking `freq` as an argument, and others have a fixed object in them, ignoring the argument.

In the case of normal objects, this just goes through without complaint:

```supercollider
p = Point(0, 3);
p.x(7)
p.x(where:7)
```

But my comment was mainly about optimization: if you call `(x:600).x(where:7)` and there is an object (here 600) is fixed, there is no need to go through the whole `makePerformableArray`.

But this is just something to keep in mind, I suppose.

Reply:
Actually I think this bit of the code is wrong because right now it only works with functions, and won't work with other classes as it just looks up the `def`. I'll change this now.